compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1984,7 +1984,11 @@ object Types {
      *  this method handles this by never simplifying inside a `MethodicType`,
      *  except for replacing type parameters with associated type variables.
      */
-    def simplified(using Context): Type = TypeOps.simplify(this, null)
+    def simplified(using Context): Type =
+      // stripping LazyRef is important for the reduction of applied match types
+      // see the comment in matchCases/recur for more details
+      val tp = stripLazyRef
```

Comment:
I would move it in TypeOps.simplify

Reply:
Recursively I think is a problem but I'll check

Reply:
Yeah, doesn't work:

```
27 suites passed, 3 failed, 30 total
    tests/neg/i15158.scala failed
    tests/neg-custom-args/isInstanceOf/i17435.scala failed
    tests/neg-custom-args/allow-deep-subtypes/i5877.scala failed
    tests/pos/i6146.scala failed
    tests/pos/i11064.scala failed
    tests/pos/i10900.scala failed
    tests/pos/i6146.scala failed
    tests/pos/i11064.scala failed
    tests/pos/i10900.scala failed
 ```